radio/src/gui/colorlcd/model_select.cpp

```diff
@@ -437,7 +439,27 @@ void ModelsPageBody::deleteModel(ModelCell *model)
       });
 }
 
-void ModelsPageBody::editLabels(ModelCell* model)
+void ModelsPageBody::saveAsTemplate(ModelCell *model)
+{
+  storageDirty(EE_MODEL);
+  storageCheck(true);
+  constexpr size_t size = sizeof(model->modelName) + sizeof(YAML_EXT);
+  char modelName[size];
+  snprintf(modelName, size, "%s%s", model->modelName, YAML_EXT);
+  char templatePath[FF_MAX_LFN];
+  snprintf(templatePath, FF_MAX_LFN, "%s%c%s", PERS_TEMPL_PATH, '/', modelName);
+  sdCheckAndCreateDirectory(TEMPLATES_PATH);
+  sdCheckAndCreateDirectory(PERS_TEMPL_PATH);
+  if (isFileAvailable(templatePath)) {
+    new ConfirmDialog(parent, STR_FILE_EXISTS, STR_ASK_OVERWRITE, [=] {
+      sdCopyFile(model->modelFilename, MODELS_PATH, modelName, PERS_TEMPL_PATH);
+    });
+  } else {
+    sdCopyFile(model->modelFilename, MODELS_PATH, modelName, PERS_TEMPL_PATH);
+  }
+}
```

Comment:
The readability could improve a lot with a few empty lines to separate blocks.
I personally would to the storageDirty.. stuff at the end, because that is the order in which this is all executed.

Reply:
This is code I copied from the fix in #2628 to re-add save as template. I am new to LVGL and have no idea what storageDirty() or storageCheck() do so not sure what I should do.  Currently works as is, may be inefficient though.  If you want, I could move those two calls to the end of that code block and see if it works.

Reply:
Those calls are nothing LVGL related, but tell the firmware that the settings have to be saved again. As I said, this is more a "I would do it different" thing than a requirement to change something.

Reply:
I moved those two calls to end of function and added a few empty lines to hopefully make it a bit more readable.  All compiles and seems to run just as well, so I committed and pushed those two files.

Reply:
Thank you very much, this looks much better.